feature/experimental/p4rt/ate_tests/google_discovery_protocol_packetin_test/google_discovery_protocol_packetin_test.go

```diff
@@ -0,0 +1,281 @@
+	component := telemetry.Component{}
+	component.IntegratedCircuit = &telemetry.Component_IntegratedCircuit{}
+	component.Name = ygot.String(*p4rtNodeName)
+	component.IntegratedCircuit.NodeId = ygot.Uint64(deviceId)
+	dut.Config().Component(*p4rtNodeName).Replace(t, &component)
+}
+
+// configurePortId configures p4rt port-id on the DUT.
+func configurePortId(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice) {
+	ports := sortPorts(dut.Ports())
+	for i, port := range ports {
+		dut.Config().Interface(port.Name()).Id().Replace(t, uint32(i)+portId)
+	}
+}
+
+// setupP4RTClient sends client arbitration message for both leader and follower clients,
+// then sends setforwordingpipelineconfig with leader client.
+func setupP4RTClient(ctx context.Context, t *testing.T, args *testArgs) error {
+	// Setup p4rt-client stream parameters
+	streamParameter := p4rt_client.P4RTStreamParameters{
+		Name:        streamName,
+		DeviceId:    deviceId,
+		ElectionIdH: uint64(0),
+		ElectionIdL: electionId,
+	}
+
+	// Send ClientArbitration message on both p4rt leader and follower clients.
+	clients := []*p4rt_client.P4RTClient{args.leader, args.follower}
+	for index, client := range clients {
+		if client != nil {
+			client.StreamChannelCreate(&streamParameter)
+			if err := client.StreamChannelSendMsg(&streamName, &p4_v1.StreamMessageRequest{
+				Update: &p4_v1.StreamMessageRequest_Arbitration{
+					Arbitration: &p4_v1.MasterArbitrationUpdate{
+						DeviceId: streamParameter.DeviceId,
+						ElectionId: &p4_v1.Uint128{
+							High: streamParameter.ElectionIdH,
+							Low:  streamParameter.ElectionIdL - uint64(index),
+						},
+					},
+				},
+			}); err != nil {
+				t.Logf("There is error when setting up p4rt client")
+				return err
+			}
+			_, _, arbErr := client.StreamChannelGetArbitrationResp(&streamName, 1)
+
+			if arbErr != nil {
+				t.Logf("There is error at Arbitration time: %v", arbErr)
+				return arbErr
+			}
+		}
+	}
+
+	// Load p4info file.
+	p4Info, err := utils.P4InfoLoad(p4InfoFile)
+	if err != nil {
+		t.Logf("There is error when loading p4info file")
+		return err
+	}
+
+	// Send SetForwardingPipelineConfig for p4rt leader client.
+	err = args.leader.SetForwardingPipelineConfig(&p4_v1.SetForwardingPipelineConfigRequest{
+		DeviceId:   deviceId,
+		ElectionId: &p4_v1.Uint128{High: uint64(0), Low: electionId},
+		Action:     p4_v1.SetForwardingPipelineConfigRequest_VERIFY_AND_COMMIT,
+		Config: &p4_v1.ForwardingPipelineConfig{
+			P4Info: &p4Info,
+			Cookie: &p4_v1.ForwardingPipelineConfig_Cookie{
+				Cookie: 159,
+			},
+		},
+	})
+	if err != nil {
+		t.Logf("There is error seen when setting SetForwardingPipelineConfig")
+		return err
+	}
+	return nil
+}
+
+// getGDPParameter returns GDP related parameters for testPacketIn testcase.
+func getGDPParameter(t *testing.T) PacketIO {
+	return &GDPPacketIO{
+		PacketIOPacket: PacketIOPacket{
+			SrcMAC:       ygot.String("00:01:00:02:00:00"),
```

Comment:
Can we add the SrcMAC also as a variable to keep it uniform?

Reply:
sure. I will make the change to convert it to variable.